app/controllers/new_threads_controller.rb

```diff
@@ -53,10 +54,12 @@ def create
 
   def edit
     @new_thread = NewThread.friendly.find(params[:id])
+    authorize! :edit, @new_thread
   end
```

Comment:
I hope you read the wiki I pointed. Under [custom find](https://github.com/CanCanCommunity/cancancan/wiki/Authorizing-controller-actions#custom-find) they explain how you can customize load.
If you want you can drop `load_resource` all together as you have done and write your own custom load method. Only important thing is that `@new_thread` should be available before call to `authorize_resource` is made. `authorize_resource` being on top of class, it is first thing which is called and by that time `@new_thread` is not available. Write a before filter which finds `@new_thread` before call to `authorize_source` is made. This would also mean you can get rid of some code duplication ( multiple use of same line  `@new_thread = NewThread.friendly.find(params[:id])` )
However first try to customize `load_resource` and see if you can get it to work.